### PR TITLE
Arm wrestle

### PR DESCRIPTION
**Pull request by aprilrieger**
Ref Ticket: https://github.com/scientist-softserv/dev-ops/issues/498

- separates the web/worker/solr builds
- and separates the platform builds arm64/amd64
- which allows our tests, lint, and deploys to happen before the arm64 image is even built and pushed to the registry since the lint/tests only depend on the amd64 build (which is roughly 15 minutes), and the deploy depends on the sha that is pushed to the registry form the amd64 build and push.

### CODE REVIEW

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -55,7 +55,7 @@ jobs:
     steps:
       - id: setup
         name: Setup
-        uses: scientist-softserv/actions/setup-env@main
+        uses: scientist-softserv/actions/setup-env@arm-wrestle
```
Comment (orangewolf) [inline .github/workflows/deploy.yaml:58]: why the deploy changes?

**Review by orangewolf — APPROVED** [review]
this looks great. the worker won't get the benefit of the web build in this version, but maybe that's ok?

File: .github/workflows/deploy.yaml (hunk 1 above)
Comment (aprilrieger) [reply .github/workflows/deploy.yaml:58]: Residual from testing. Removing shortly.

**Comment by aprilrieger** [comment]
What would the benefit be? Caching?

**Pull request merged**